storage/memory/memory.go

```diff
@@ -219,22 +216,22 @@ func (s *MemoryBackend) ReadChanges(ctx context.Context, store, objectType strin
 	return res, []byte(continuationToken), nil
 }
 
-func (s *MemoryBackend) read(ctx context.Context, store string, key *openfgapb.TupleKey, paginationOptions storage.PaginationOptions) (*staticIterator, error) {
+func (s *MemoryBackend) read(ctx context.Context, store string, tk *openfgapb.TupleKey, paginationOptions storage.PaginationOptions) (*staticIterator, error) {
 	_, span := s.tracer.Start(ctx, "memory.read")
 	defer span.End()
 
 	s.mu.Lock()
 	defer s.mu.Unlock()
 
-	if key.Object == "" && key.User == "" {
-		err := errObjectOrUserMustBeSpecified
-		telemetry.TraceError(span, err)
-		return nil, err
-	}
 	var matches []*openfgapb.Tuple
-	for _, t := range s.tuples[store] {
-		if match(key, t.Key) {
-			matches = append(matches, t)
+	if tk.GetObject() == "" && tk.GetRelation() == "" && tk.GetUser() == "" {
+		matches = make([]*openfgapb.Tuple, len(s.tuples[store]))
+		copy(matches, s.tuples[store])
```

Comment:
Instead of `copy` why not just return a static iterator over the `s.tuples[store]` that you already have?

Reply:
Because of the way that the pagination code works below, this way seems like the most straightforward.

Reply:
Sounds good 👍